src/client/features/enrichment/token-input.js

```diff
@@ -0,0 +1,80 @@
+const React = require('react');
+const h = require('react-hyperscript');
+const _ = require('lodash');
+const { ServerAPI } = require('../../services/');
+let Textarea = require('react-textarea-autosize').default;
+
+class TokenInput extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: '',
+      validTokens: [],
+      unrecognizedTokens: []
+    };
+  }
+
+  //store 'gene-input-box' contents on state
+  handleChange(e) {
+    this.state.query = e.target.value;
+  }
+
+  //call validation service API to retrieve validation result in the form of []
+  retrieveValidationAPIResult(){
+    //reset state values
+    this.state.unrecognizedTokens = [];
+    this.state.validTokens = [];
```

Comment:
do not modify state directly (https://reactjs.org/docs/state-and-lifecycle.html#do-not-modify-state-directly)